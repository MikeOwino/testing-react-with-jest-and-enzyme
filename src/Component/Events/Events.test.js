import React from 'react'
import { shallow } from 'enzyme'
import Events from './Events'

describe('Events Component', () => {

    let wrapper;
    beforeEach(() => {
        wrapper = shallow(<Events />);
    });

    test("Should render a button with `increment`", () => {
      let increment = wrapper.find(`[data-test='increment-btn']`).text();
      expect(increment).toBe('Increment');
    });

    test("Should render the initial value 0", () => {
       let counter = wrapper.find(`[data-test='counter-value']`).text();
       expect(counter).toBe("0");
    });

    test("Should render the click event of increment button", () => {
       wrapper.find(`[data-test='increment-btn']`).simulate("click"); 
       expect(wrapper.find(`[data-test='counter-value']`).text()).toBe("1");
    });

    test("Should render the click event of decrement button", () => {
       wrapper.find(`[data-test='decrement-btn']`).simulate("click");
       expect(wrapper.find(`[data-test='counter-value']`).text()).toBe("0");
    });

});
