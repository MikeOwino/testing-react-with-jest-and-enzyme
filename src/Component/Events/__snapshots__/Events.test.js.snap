// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Events Component Matches the snapshot 1`] = `
<div
  className="Counter"
>
  <h1
    data-test="counter-title"
  >
    Event & Snapshot Test
  </h1>
  <div
    data-test="counter-value"
  >
    0
  </div>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
    data-test="increment-btn"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Increment
    </span>
  </button>
  <br />
   
  <br />
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
    data-test="decrement-btn"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Decrement
    </span>
  </button>
</div>
`;
